Xamarin.Forms.Xaml.UnitTests/XamlC/TypeReferenceExtensionsTests.cs

```diff
@@ -118,5 +126,20 @@ public void TestSameTypeNamesFromDifferentAssemblies()
 
 			Assert.False(TestInheritsFromOrImplements(test.GetType("Xamarin.Forms.Effect"), core.GetType("Xamarin.Forms.Effect")));
 		}
+
+		[Test]
+		public void TestImplementsGenericInterface()
```

Comment:
can't you replace that test by a `TestCase` for `TestInheritsFromOrImplements` ?

Reply:
No because `ImplementsGenericInterface` does not share implementation with `InheritsFromOrImplements`.

By the way, the review gave me idea that the implementation of the generic parameter resolution, `ResolveGenericParameter` can be shared. I did that with commit 1b697b17ddaabe70a6e07169f626a038a29346e2.